lib/validators/specValidator.js

```diff
@@ -64,23 +64,6 @@ class SpecValidator {
     if (options.shouldResolveRelativePaths === null || options.shouldResolveRelativePaths === undefined) {
       options.shouldResolveRelativePaths = true;
     }
-    if (options.shouldResolveXmsExamples === null || options.shouldResolveXmsExamples === undefined) {
-      options.shouldResolveXmsExamples = true;
-    }
-    if (options.shouldResolveAllOf === null || options.shouldResolveAllOf === undefined) {
-      options.shouldResolveAllOf = true;
-    }
-    if (options.shouldSetAdditionalPropertiesFalse === null || options.shouldSetAdditionalPropertiesFalse === undefined) {
-      options.shouldSetAdditionalPropertiesFalse = true;
-    }
-    if (options.shouldResolvePureObjects === null || options.shouldResolvePureObjects === undefined) {
-      options.shouldResolvePureObjects = true;
-    }
-    if (options.shouldResolveDiscriminator === null || options.shouldResolveDiscriminator === undefined) {
-      // Resolving allOf is a neccessary precondition for resolving discriminators. Hence hard setting this to true
-      options.shouldResolveAllOf = true;
-      options.shouldResolveDiscriminator = true;
-    }
```

Comment:
removing the above conditions as they are being validated in the specResolver constructor. We are not exposing the resolver. It is accessed by the validator. SpecValidator is a pass thru.